### PR TITLE
feat(query): support explain insert with insert source

### LINKED ISSUE #14765: Feature:  improve explain insert statement
**Summary**

Description for this feature.

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

support explain insert with insert source

- Fixes #14765 

## Tests

- [x] Logic Test

## Type of change

- [x] New Feature (non-breaking change which adds functionality)

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15357)

### CODE REVIEW

File: src/query/service/src/interpreters/interpreter_explain.rs [hunk 1]
```diff
@@ -557,6 +542,161 @@ impl ExplainInterpreter {
...
+                }
+                InsertValue::RawValues { .. } => {
+                    formatted_plan = FormatTreeNode::with_children(
+                        "InsertPlan (rawvalues):".to_string(),
+                        children,
+                    )
+                    .format_pretty()?;
+                }
+            },
+            InsertInputSource::Stage(plan) => match *plan.clone() {
```
Comment (sundy-li) [inline src/query/service/src/interpreters/interpreter_explain.rs:621]: better move these codes into plan/insert.

Comment (TCeason) [reply src/query/service/src/interpreters/interpreter_explain.rs]: Done

File: src/query/sql/src/planner/plans/insert.rs [hunk 2]
```diff
@@ -82,6 +87,152 @@ impl Insert {
...
+        let mut formatted_plan = String::new();
+
+        match source {
+            InsertInputSource::SelectPlan(plan) => {
+                if let Plan::Query {
+                    s_expr, metadata, ..
+                } = &**plan
+                {
+                    let metadata = &*metadata.read();
+                    let sub_tree = s_expr.to_format_tree(metadata, verbose)?;
```
Comment (Dousir9) [inline src/query/sql/src/planner/plans/insert.rs:132]: It’s would be better to display physical plan here, but due to the lack of `BindContext`, it is not easy to do it, I will change it to physical plan after refactoring Binder.

**Review by Dousir9 — APPROVED** [review]

File: src/query/sql/src/planner/plans/insert.rs (hunk 2 above)
Comment (TCeason) [reply src/query/sql/src/planner/plans/insert.rs:132]: Mark

**Review by sundy-li — APPROVED** [review]

**Comment by TCeason** [comment]
cc @BohuTANG  Please merge this pr.

**Pull request merged**